loader/loader.go

```diff
@@ -479,18 +482,46 @@ func (l *Loader) protoEnum(tspec *ast.TypeSpec) (*desc.EnumDescriptorProto, erro
 	return enum, nil
 }
 
-func (l *Loader) protoType(expr ast.Expr, pkg *types.Package) (desc.FieldDescriptorProto_Type, string) {
+func (l *Loader) protoType(expr ast.Expr, pkg *types.Package) (desc.FieldDescriptorProto_Type, desc.FieldDescriptorProto_Label, string) {
 	if pkg == nil {
 		pkg = l.tpkg
 	}
 	switch x := expr.(type) {
 	case *ast.Ident:
+		// Map Go types to proto types
+		// https://developers.google.com/protocol-buffers/docs/proto3#scalar
 		switch x.Name {
 		case "string":
-			return desc.FieldDescriptorProto_TYPE_STRING, x.Name
+			return desc.FieldDescriptorProto_TYPE_STRING, 0, x.Name
+		case "int", "int32":
+			return desc.FieldDescriptorProto_TYPE_INT32, 0, x.Name
+		case "bool":
+			return desc.FieldDescriptorProto_TYPE_BOOL, 0, x.Name
 		default:
 			fullName := "." + pkg.Path() + "." + x.Name
-			return desc.FieldDescriptorProto_TYPE_ENUM, fullName
+			// Check if the identifier is an already known 'message' or 'enum.
+			// NOTE: This checks all of the parsed gunk packages for the type.
+			// TODO: There is likely a much more robust way to handle this.
+			for _, p := range l.astPkgs[pkg.Path()] {
+				for _, decl := range p.Decls {
+					gd := decl.(*ast.GenDecl)
```

Comment:
This will panic if a package happenst to have a top-level func declaration. Use the `, ok` form, and skip if `!ok`.

Reply:
If the old code did `gd, _ := decl.(*ast.GenDecl)` then perhaps, but if you use the non-bool form, the expression will panic instead of giving you nil. Thanks for fixing it :)

Reply:
Ah sorry, I was referring to the function that calls this (rather than old code) - https://github.com/gunk/gunk/blob/master/loader/loader.go#L236